### ISSUE TITLE
Update the GraphHooks to make them all json serialiseable

### ISSUE THREAD

**Issue by p013570** [gchq/Gaffer#1089]
Each of our core graph hooks are instantiated in different ways, loading data from config files in different ways. It would be useful to standardise this and make the process simpler. Instead of having lots of config files for each of the different graph hooks, I think it would be neater to have 1 json file containing all the configuration for all your graph hooks. Something similar to this:

```json
[
  {
    "class": "uk.gov.gchq.gaffer.graph.hook.OperationChainLimiter",
    "opScores": {
      "uk.gov.gchq.gaffer.operation.Operation": 1,
      "uk.gov.gchq.gaffer.operation.impl.add.AddElements": 2,
      "uk.gov.gchq.gaffer.operation.impl.get.GetAllElement": 5,
      "uk.gov.gchq.gaffer.operation.impl.generate.GenerateObjects": 0
    },
    "authScores": {
        "User": 2,
        "SuperUser": 5
    }
  },
  {
    "class": "uk.gov.gchq.gaffer.graph.hook.AddOperationsToChain",
    "end": [
      {
        "class": "uk.gov.gchq.gaffer.operation.impl.Count"
      }
    ],
    "after": {
      "uk.gov.gchq.gaffer.operation.impl.get.GetAllElements": [
        {
          "class": "uk.gov.gchq.gaffer.operation.impl.Limit"
        }
      ]
    }
  },
  {
    "class": "uk.gov.gchq.gaffer.graph.hook.OperationAuthoriser",
    "auths": {
      "uk.gov.gchq.gaffer.operation.Operation": [
        "User"
      ],
      "uk.gov.gchq.gaffer.operation.io.Output": [
        "ReadUser"
      ],
      "uk.gov.gchq.gaffer.operation.impl.add.AddElements": [
        "WriteUser"
      ],
      "uk.gov.gchq.gaffer.operation.impl.get.GetAdjacentIds": [
        "SuperUser"
      ],
      "uk.gov.gchq.gaffer.operation.impl.get.GetAllElements": [
        "AdminUser",
        "SuperUser"
      ]
    }
  }
]
```

We should this json annotation to the GraphHook class:
```java
@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "class")
```

We should also add a unit test for GraphHook implementations to test it can be serialised/deserialised to json.

**Comment by p013570** [comment]
#Migration Steps
Graph hooks are now configured in json. This simplifies the configuration required to add graph hooks to a Gaffer rest api. You can now provide one json files containing the configuration for all of your graph hooks. For example:

```json
[
  {
    "class": "uk.gov.gchq.gaffer.graph.hook.OperationChainLimiter",
    "opScores": {
      "uk.gov.gchq.gaffer.operation.Operation": 1,
      "uk.gov.gchq.gaffer.operation.impl.add.AddElements": 2,
      "uk.gov.gchq.gaffer.operation.impl.get.GetAllElement": 5,
      "uk.gov.gchq.gaffer.operation.impl.generate.GenerateObjects": 0
    },
    "authScores": {
        "User": 2,
        "SuperUser": 5
    }
  },
  {
    "class": "uk.gov.gchq.gaffer.graph.hook.AddOperationsToChain",
    "end": [
      {
        "class": "uk.gov.gchq.gaffer.operation.impl.Count"
      }
    ],
    "after": {
      "uk.gov.gchq.gaffer.operation.impl.get.GetAllElements": [
        {
          "class": "uk.gov.gchq.gaffer.operation.impl.Limit"
        }
      ]
    }
  },
  {
    "class": "uk.gov.gchq.gaffer.graph.hook.OperationAuthoriser",
    "auths": {
      "uk.gov.gchq.gaffer.operation.Operation": [
        "User"
      ],
      "uk.gov.gchq.gaffer.operation.io.Output": [
        "ReadUser"
      ],
      "uk.gov.gchq.gaffer.operation.impl.add.AddElements": [
        "WriteUser"
      ],
      "uk.gov.gchq.gaffer.operation.impl.get.GetAdjacentIds": [
        "SuperUser"
      ],
      "uk.gov.gchq.gaffer.operation.impl.get.GetAllElements": [
        "AdminUser",
        "SuperUser"
      ]
    }
  }
]
```

## Migration notes for individual Graph Hooks

### OperationAuthoriser
Instead of supplying the opAuths.properties file, you now need to configure the hook in json using the auths field. auths is a map of operation class name to array of string authorisations. See above for a json example.

### OperationChainLimiter (and ScoreOperationChainHandler)
Rather than providing the authScores.properties and opScores.properties files you now need to configure the hook in json using the authScores and opScores fields. See above for a json example.

- authScores is a map of auth string to integer score.
- opScores is a map of operation class name to integer score.

The ScoreOperationChainHandler score property files have also been removed and replaced with a single json file similar to OperationChainLimiter. The only difference with the json file is that it does not have a class name field.

For example:

ScoreOperationChainDeclaration.json:
```json
{
  "operations": [
    {
      "operation": "uk.gov.gchq.gaffer.operation.impl.ScoreOperationChain",
      "handler": {
        "class": "uk.gov.gchq.gaffer.store.operation.handler.ScoreOperationChainHandler",
        "opScores": {
          "uk.gov.gchq.gaffer.operation.Operation": 1,
          "uk.gov.gchq.gaffer.operation.impl.generate.GenerateObjects": 0
        },
        "authScores": {
          "TEST_USER": 4,
          "TEST_USER_ENHANCED": 10
        }
      }
    }
  ]
}

```

## REST API changes
The system properties have been removed:
- gaffer.operation.auths.path
- gaffer.graph.hook.add.operations.path
- gaffer.chain.limiter.enable
- gaffer.chain.limiter.operation.scores.path
- gaffer.chain.limiter.authorisation.scores.path

They have been replaced with a single property:
- gaffer.graph.hooks.path
this should be set to a path to your json file containing an array of graph hooks as shown above.

**Comment by p013570** [comment]
Merged into develop-0.8.0
